static/app/views/alerts/rules/metric/eapField.tsx

```diff
@@ -0,0 +1,196 @@
+        }).includes(metric.name)
+      );
+  }, [meta]);
+
+  // We parse out the aggregation and field from the aggregate string.
+  // This only works for aggregates with <= 1 argument.
+  const {
+    name: aggregation,
+    arguments: [field],
+  } = parseFunction(aggregate) ?? {arguments: [undefined]};
+
+  useEffect(() => {
+    const selectedMriMeta = metaArr.find(metric => metric.name === field);
+    if (field && !selectedMriMeta && !isLoading) {
+      const newSelection = metaArr[0];
+      if (newSelection) {
+        onChange(`count(${newSelection.name})`, {});
+      } else if (aggregate !== DEFAULT_EAP_METRICS_ALERT_FIELD) {
+        onChange(DEFAULT_EAP_METRICS_ALERT_FIELD, {});
+      }
+    }
+  }, [metaArr, onChange, isLoading, aggregate, aggregation, field]);
+
+  const handleFieldChange = useCallback(
+    option => {
+      const selectedMeta = metaArr.find(metric => metric.name === option.value);
+      if (!selectedMeta) {
+        return;
+      }
+      onChange(`${aggregation}(${option.value})`, {});
+    },
+    [metaArr, onChange, aggregation]
+  );
+
+  // As SelectControl does not support an options size limit out of the box
+  // we work around it by using the async variant of the control
+  const getFieldOptions = useCallback(
+    (searchText: string) => {
+      const filteredMeta = metaArr.filter(
+        ({name}) =>
+          searchText === '' || name.toLowerCase().includes(searchText.toLowerCase())
+      );
+
+      const options = filteredMeta.splice(0, 100).map<{
+        label: React.ReactNode;
+        value: string;
+        disabled?: boolean;
+        trailingItems?: React.ReactNode;
+      }>(metric => ({
+        label: middleEllipsis(metric.name, 50, /\.|-|_/),
+        value: metric.name,
+        trailingItems: (
+          <Fragment>
+            <Tag tooltipText={t('Type')}>{getReadableMetricType(metric.type)}</Tag>
+            <Tag tooltipText={t('Unit')}>{metric.unit}</Tag>
+          </Fragment>
+        ),
+      }));
+
+      if (filteredMeta.length > options.length) {
+        options.push({
+          label: (
+            <SizeLimitMessage>{t('Use search to find more options…')}</SizeLimitMessage>
+          ),
+          value: '',
+          disabled: true,
+        });
+      }
+      return options;
+    },
+    [metaArr]
+  );
+
+  // When using the async variant of SelectControl, we need to pass in an option object instead of just the value
+  const selectedOption = field && {
+    label: field,
+    value: field,
+  };
+
+  return (
+    <Wrapper>
+      <StyledSelectControl
+        searchable
+        isDisabled={isLoading}
+        placeholder={t('Select an operation')}
+        options={OPERATIONS}
+        value={aggregation}
+        onChange={option => {
+          if (field) {
+            onChange(`${option.value}(${field})`, {});
+          } else {
+            onChange(`${option.value}(${DEFAULT_EAP_FIELD})`, {});
+          }
+        }}
```

Comment:
Probably want to wrap this in a `useCallback` to avoid excessive re-renders on the `StyledSelectControl` component